interactive/dashboard/api.py

```diff
@@ -0,0 +1,1036 @@
+        newData = json.loads(request.body)
+        path = os.path.join(settings.BASE_DIR,
+                             'configs/'
+                             +request.session.session_key
+                             +"/initials.json")
+        log("saving to path: "+path)
+        with open(path, 'w') as f:
+            json.dump(newData, f, indent=4)
+        log('initial conditions setup updated')
+        return JsonResponse({})
+
+
+class InitialSpeciesConcentrations(views.APIView):
+    def get(self, request):
+
+        log("****** GET request received INITIAL SPECIES CONCENTRATIONS ******")
+        if not request.session.session_key:
+            request.session.create()
+
+        log("fetching initial species conc. for session id: " +
+              request.session.session_key)
+        direc = os.path.join(settings.BASE_DIR,
+                                'configs/'
+                                +request.session.session_key
+                                +"/species.json")
+        if not os.path.isfile(direc):
+            log("detected no data from this user")
+            return JsonResponse({})
+        else:
+
+            path = os.path.join(settings.BASE_DIR, 'configs/' +
+                                request.session.session_key+"/species.json")
+            log("getting initial species concentrations:", path)
+            values = initial_species_concentrations(path)
+            log("returning species [isc]:", values)
+            return JsonResponse(values)
+
+    def post(self, request):
+        log("****** POST request received INITIAL SPECIES CONC. ******")
+        if not request.session.session_key:
+            request.session.create()
+        log("received options:", request.body)
+        initial_values = json.loads(request.body)
+        path = os.path.join(settings.BASE_DIR, 'configs/' +
+                            request.session.session_key+"/species.json")
+        log("saving to path: "+path)
+        if not os.path.isfile(path):
+            log("* detected no species this user")
+            return Response(status=status.HTTP_200_OK)
+        else:
+            log("* pushing new options:", initial_values)
+            formulas = {}
+            units = {}
+            values = {}
+            i = 0
+            for key, value in initial_values.items():
+                name = "Species " + str(i)
+                formulas[name] = key
+                units[name] = value["units"]
+                values[name] = value["value"]
+                i += 1
+            file_data = {}
+            file_data["formula"] = formulas
+            file_data["unit"] = units
+            file_data["value"] = values
+            with open(path, 'w') as f:
+                json.dump(file_data, f, indent=4)
+            # run export next?
+            export_to_path(os.path.join(settings.BASE_DIR,
+                           'configs/'+request.session.session_key)+"/")
+            return JsonResponse({})
```

Comment:
What's the logic for returning an empty `JsonResponse` as opposed to a `Response(status=status.HTTP_200_OK)` throughout these requests?

Reply:
So you're saying that returning another `Response(status=status.HTTP_200_OK)` would cause the page to refresh? That doesn't make any sense.

Reply:
Future work for #291